src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+/**
+ * @author Aurimas Niekis <aurimas@niekis.lt>
+ */
+class RedisAdapter extends AbstractAdapter
+{
+    /**
+     * @var \Redis
+     */
+    private $redis;
+
+    /**
+     * @param \Redis $redisConnection
+     * @param string $namespace
+     * @param int    $defaultLifetime
+     */
+    public function __construct($redisConnection, $namespace = '', $defaultLifetime = 0)
+    {
+        $this->redis = $redisConnection;
+
+        parent::__construct($namespace, $defaultLifetime);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        $values = $this->redis->getMultiple($ids);
+
+        $index = 0;
+        $result = [];
+
+        foreach ($ids as $id) {
+            $value = $values[$index++];
+
+            if (false === $value) {
+                continue;
+            }
+
+            $result[$id] = unserialize($value);
+        }
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->redis->exists($id);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear()
+    {
+        return $this->redis->flushDB();
```

Comment:
Redis does not support delete with prefix or etc... I could do get keys and when do foreach but you know...

+ APCU does also clear all cache...